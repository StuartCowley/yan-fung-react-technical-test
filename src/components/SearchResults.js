import React from "react";
import "../styles/search_results.css"

const SearchResults = ({ results }) => {
  if (!results?.length) {
    return <p>No results</p>
  } else {
    return (
      <div className="search-results">
        {results.map((image, index) => {
          return (
            <img 
              className="search-results__image" 
              src={image} 
              alt="space_image" 
              key={index}
            />
          )
        })}
      </div>
    )
  }
}
export default SearchResults;
