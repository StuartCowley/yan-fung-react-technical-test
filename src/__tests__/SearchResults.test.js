import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search results", () => {
  const validProps = {
    results: ["test1", "test2"]
  }

  const { asFragment } = render(<SearchResults results={validProps.results} />);

  it("render correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders `No results` if there is no result", () => {
    const validProps = [];
    render(<SearchResults results={validProps} />);
    const placeholderMessage = screen.getByText("No results");
    
    expect(placeholderMessage).toBeInTheDocument();
  })
});
