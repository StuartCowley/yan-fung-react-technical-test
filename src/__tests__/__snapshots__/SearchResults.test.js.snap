// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search results render correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="space_image"
      class="search-results__image"
      src="test1"
    />
    <img
      alt="space_image"
      class="search-results__image"
      src="test2"
    />
  </div>
</DocumentFragment>
`;
